src/lib/stores/navigation.ts

```diff
@@ -0,0 +1,36 @@
+import { resolve } from '$app/paths';
+import { goto } from '$app/navigation';
+import type { Pathname, RouteId, RouteParams } from '$app/types';
+
+// taken directly from svelte's source!
+type ResolveArgs<T extends RouteId | Pathname> = T extends RouteId
+    ? RouteParams<T> extends Record<string, never>
+        ? [route: T]
+        : [route: T, params: RouteParams<T>]
+    : [route: T];
+
+export function withPath(base: string, ...parts: string[]) {
+    // remove slashes at the end if any
+    const normalizedBase = base.replace(/\/+$/, '');
+
+    // remove slashes at the start of each part if any
+    const normalizedParts = parts.map((part) => part.replace(/^\/+/, ''));
+
+    // join em with slashes
+    return [normalizedBase, ...normalizedParts].join('/');
+}
+
+export function resolveRoute<T extends RouteId>(route: T, params?: Record<string, string>) {
+    // type cast is necessary here!
+    const resolveArgs = params ? ([route, params] as [T, RouteParams<T>]) : [route];
+
+    return resolve(...(resolveArgs as ResolveArgs<T>));
+}
```

Comment:
why cant we just use the native resolve?

Reply:
@TorstenDittmann 
it doesn't allow doing something like - `resolve(..., page.params)` because apparently everything in `page.params` is nullable and you have to explicity use an object like -
```ts
resolve('...', { 
  project: page.params.project,
  region: page.params.region 
});
```
this method allows use to simply do - 
```ts
resolve('...', page.params);
```
This is more of a sugary syntax helper 😄